### ISSUE TITLE
Add docker label-schema.org support

### ISSUE THREAD

**Issue by docwhat** [goreleaser/goreleaser#813]
It would be nice if the `dockers` section would have support either for arbitrary `docker build` flags, or explicit support for <http://label-schema.org/rc1/> (or both).

One example might be something like:

``` yaml
dockers:
  build-flags:
  - "--label=org.label-schema.schema-version=1.0"
  - "--label=org.label-schema.vcs-ref={{.Commit}}"
  - "--label=org.label-schema.name={{.ProjectName}}"
  - "--build-arg=FRED={{.Tag}}"
```

You get the idea.

**Comment by caalberts** [comment]
i'd like to give this a shot

**Comment by caarlos0** [comment]
Great! Let me know if you need any help :)

**Comment by caalberts** [comment]
I'd like to break this down a little bit more. There are 2 different sets of requirements here:
1. support arbitrary `docker build` flags (including any label)
2. support labels following label-schema convention.

The first can simply be `build-flags` as suggested. For the second, I'm thinking of a separate property `label-schema`, which makes it easier to validate the presence of `schema-version` and also takes care of prefixing the labels.

Something like:

```yml
dockers:
  
  build-flags:
  - "--build-arg=FRED={{.Tag}}"
  - "--label=foo=bar"
  
  label-schema:
    schema-version: "1.0"
    name: "{{.ProjectName}}"
    version: "1.2.3"
    vcs-ref: "{{.Commit}}"
```

What do you think?

**Comment by docwhat** [comment]
I like! In theory, just the `build-flags` are needed.

**Comment by caarlos0** [comment]
Yeah, I think we can keep just the build flags, with template support, and that's it... 

If we add both, we'll have to deal with possible duplicated flags and things like that...

**Comment by docwhat** [comment]
Yup.

**Comment by caalberts** [comment]
> Yeah, I think we can keep just the build flags, with template support, and that's it...

👍

**Comment by caarlos0** [comment]
fixed on #819
